app/models/manageiq/providers/embedded_ansible/automation_manager/credential.rb

```diff
@@ -26,6 +26,18 @@ def self.raw_create_in_provider(manager, params)
     create!(create_params)
   end
 
+  def self.password_attribute_keys
+    self::API_ATTRIBUTES.map do |k, v|
```

Comment:
Since `API_ATTRIBUTES` are mapped to different DB columns as part of the `params_to_attributes`, maybe include `encrypted_columns` from `PasswordMixin` to this list before mapping?

Reply:
Yeah, I did this specifically for the API attributes. For other attributes you could use `.encrypted_columns`. I can change the name to be more clear. Maybe `api_attribute_password_keys`?

Reply:
I guess what I'm saying is if the API attributes and the encrypted columns were the same then I wouldn't need this method.

Since that's the case then I wouldn't want to confuse the two. I'm not sure what the benefit of having both here would be.

Reply:
I was only just suggesting this in the off chance something else besides the UI ends up queuing a credential operation, and isn't using the `API_ATTRIBUTE` keys to do it.

Not a big deal, so I don't think this is worth stopping a merge.

Reply:
I don't think that would even work. Pretty sure `params_to_attributes` would break the cred data (we assume that the input has the `API_ATTRIBUTES`-style keys)

Reply:
Okay, I concede...

https://github.com/ManageIQ/manageiq/blob/712d9cd16187282f08d64f3016ae3abc66f6e621/app/models/manageiq/providers/embedded_ansible/automation_manager/vault_credential.rb#L29

Though I had done a `||=` in some places, but I guess not.


You win this time Carboni! (_shakes fist in air while onlookers look on in wonder..._)